backend/src/dao/SignInFormDao.ts

```diff
@@ -10,22 +10,24 @@ export default class SignInFormDao {
     if (!formRef.exists) throw new NotFoundError(`No form with id '${id}' found.`);
     const form = formRef.data();
     if (form == null) throw new NotFoundError(`No form content in form with id '${id}' found.`);
+    const signedInAtVal = Date.now();
+    if (form.expireAt <= signedInAtVal)
+      throw new PermissionError(`User is not allowed to sign into expired form with id '${id}.`);
```

Comment:
I think it might be better to have this as `BadRequestError`, doesn't seem to quite fit a permissions error.

Reply:
Done!